test/integration/controlplane/transformation/secrets_transformation_test.go

```diff
@@ -431,19 +433,26 @@ func TestListCorruptObjects(t *testing.T) {
 		// secrets that are created before encryption breaks
 		secrets []string
 		// whether encryption broke after the config change
-		encryptionBrokenFn func(t *testing.T, got apierrors.APIStatus) bool
+		encryptionBrokenFn func(t *testing.T, got apierrors.APIStatus)
 		// what we expect for LIST on the corrupt objects after encryption has broken
 		listAfter verifier
 	}{
 		{
 			secrets:        secrets,
 			featureEnabled: true,
-			encryptionBrokenFn: func(t *testing.T, got apierrors.APIStatus) bool {
-				// the new encryption config does not have the old key, so reading of resources
-				// created before the encryption change will fail with 'no matching prefix found'
-				return got.Status().Reason == metav1.StatusReasonInternalError &&
-					strings.Contains(got.Status().Message, "Internal error occurred: StorageError: corrupt object") &&
-					strings.Contains(got.Status().Message, "data from the storage is not transformable revision=0: no matching prefix found")
+			encryptionBrokenFn: func(t *testing.T, got apierrors.APIStatus) {
+				status := got.Status()
+				if status.Reason != metav1.StatusReasonInternalError {
+					t.Errorf("Invalid reason, got: %q, want: %q", status.Reason, metav1.StatusReasonInternalError)
+				}
+				corruptObjectMsg := "Internal error occurred: StorageError: corrupt object"
+				if !strings.Contains(status.Message, corruptObjectMsg) {
+					t.Errorf("Message should include %q, but got: %q", corruptObjectMsg, status.Message)
+				}
+				messageAuthrnticationFailedMsg := "data from the storage is not transformable revision=0: cipher: message authentication failed"
```

Comment:
Here the low level error is not the same 'no matching prefix found' is replaced with `cipher: message authentication failed`, which is expected as we no longer replace key but corrupt object.